ecloud/resource_loadbalancer_vip.go

```diff
@@ -0,0 +1,235 @@
+	d.Set("load_balancer_id", lbVip.LoadBalancerID)
+
+	
+	if d.Get("floating_ip_id").(string) == "" && d.Get("allocate_floating_ip").(bool) {
+		//we need to use the IP ID from the vip
+		//and filter for the floating IP ID
+		params := connection.APIRequestParameters{}
+		params.WithFilter(*connection.NewAPIRequestFiltering("resource_id", connection.EQOperator, []string{lbVip.IPAddressID}))
+
+		fips, err := service.GetFloatingIPs(params)
+		if err != nil {
+			return fmt.Errorf("Failed to retrieve floating IPs: %w", err)
+		}
+
+		if len(fips) != 1 {
+			return fmt.Errorf("Unexpected number of floating IPs allocated to VIP")
+		}
+
+		d.Set("floating_ip_id", fips[0].ID)
+	}
+
+	return nil
+}
+
+func resourceLoadBalancerVipUpdate(d *schema.ResourceData, meta interface{}) error {
+	service := meta.(ecloudservice.ECloudService)
+
+	if d.HasChange("name") {
+		log.Printf("[INFO] Updating loadbalancer vip with ID [%s]", d.Id())
+		patchReq := ecloudservice.PatchVIPRequest{
+			Name: d.Get("name").(string),
+		}
+
+		taskRef, err := service.PatchVIP(d.Id(), patchReq)
+		if err != nil {
+			return fmt.Errorf("Error updating loadbalancer vip with ID [%s]: %w", d.Id(), err)
+		}
+
+		stateConf := &resource.StateChangeConf{
+			Target:     []string{ecloudservice.TaskStatusComplete.String()},
+			Refresh:    TaskStatusRefreshFunc(service, taskRef.TaskID),
+			Timeout:    d.Timeout(schema.TimeoutUpdate),
+			Delay:      3 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf("Error waiting for loadbalancer vip with ID [%s] to return task status of [%s]: %s", d.Id(), ecloudservice.TaskStatusComplete, err)
+		}
+	}
+
+	return resourceLoadBalancerVipRead(d, meta)
+}
+
+func resourceLoadBalancerVipDelete(d *schema.ResourceData, meta interface{}) error {
+	service := meta.(ecloudservice.ECloudService)
+
+	log.Printf("[INFO] Removing loadbalancer network with ID [%s]", d.Id())
+	taskID, err := service.DeleteVIP(d.Id())
+	if err != nil {
+		switch err.(type) {
+		case *ecloudservice.VIPNotFoundError:
+			log.Printf("[DEBUG] loadbalancer VIP with ID [%s] not found. Continuing.", d.Id())
+		default:
+			return fmt.Errorf("Error removing loadbalancer vip with ID [%s]: %s", d.Id(), err)
+		}
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{ecloudservice.TaskStatusComplete.String()},
+		Refresh:    TaskStatusRefreshFunc(service, taskID),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for loadbalancer vip with ID [%s] to be deleted: %s", d.Id(), err)
+	}
+
+	//remove floating ip if set
+	if len(d.Get("floating_ip_id").(string)) > 1 {
+		fip := d.Get("floating_ip_id").(string)
+
+		log.Printf("[DEBUG] Unassigning floating ip with ID [%s]", fip)
+
+		taskID, err := service.UnassignFloatingIP(fip)
```

Comment:
I don't believe unassign is required here - removing the VIP will unassign the floating IP from the VIP ip address if i'm not mistaken?

Reply:
Good catch. Yeah that was from before I switched the delete ordering. Will get that cleaned up